### PR TITLE
Add TLSSocket example

### PR DESCRIPTION
**Pull request by SeppoTakalo**
This is nearly a verbatim copy of Socket examples, but now uses TLSSocket.

### CODE REVIEW

**Comment by SeppoTakalo** [comment]
@AnotherButler Please review & merge.

Will be linked into the ongoing TLSSocket documentation PR

File: TLSSocket/main.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#include "mbed.h"
+#include "mbed_trace.h"
+
+const char* cert = \
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: Wasting 4 bytes of RAM. Should be `const char cert[] =`

File: TLSSocket/main.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+DISCONNECT:
+    delete[] buffer;
+    // Close the socket to return its memory and bring down the network interface
+    socket->close();
+
+    // Bring down the ethernet interface
+    net->disconnect();
+    printf("Done\n");
+}
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: Missing newline

File: TLSSocket/main.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        if (result < 0) {
+            printf("Error! socket->send() returned: %d\n", result);
+            goto DISCONNECT;
+        }
+        size -= result;
+    }
+
+    // Receieve an HTTP response and print out the response line
+    while (0 < (result = socket->recv(buffer, 255))) {
+        buffer[255] = 0;
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: `buffer[result]`, surely?

File: TLSSocket/main.cpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    if (result != 0) {
+        printf("Error! socket->connect() returned: %d\n", result);
+        goto DISCONNECT;
+    }
+
+    // Send a simple http request
+    size = strlen(query);
+
+    // Loop until whole request sent
+    while(size) {
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: I don't /think/ this loop should be necessary. A blocking socket or file handle is supposed to send everything. (In Unix a signal could interrupt, but not mbed OS).

Missing space, anyway

File: TLSSocket/main.cpp [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            printf("Error! socket->send() returned: %d\n", result);
+            goto DISCONNECT;
+        }
+        size -= result;
+    }
+
+    // Receieve an HTTP response and print out the response line
+    while (0 < (result = socket->recv(buffer, 255))) {
+        buffer[255] = 0;
+        printf("%s\n", buffer);
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: Doesn't this add `\n` at buffer/packet boundaries?

File: TLSSocket/main.cpp [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    printf("Connecting to network\n");
+    result = net->connect();
+    if (result != 0) {
+        printf("Error! net->connect() returned: %d\n", result);
+        return result;
+    }
+
+    TLSSocket *socket = new TLSSocket;
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: As you've configured a big main stack, no particular reason to `new` this? Or the buffer?

File: TLSSocket/main.cpp [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    mbed_trace_init();
+
+    printf("TLSSocket Example.\n");
+    printf("Mbed OS version: %d.%d.%d\n\n", MBED_MAJOR_VERSION, MBED_MINOR_VERSION, MBED_PATCH_VERSION);
+
+    NetworkInterface *net = NetworkInterface::get_default_instance();
+
+    if (!net) {
+        printf("Error! No network inteface found.\n");
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: interface

File: TLSSocket/main.cpp [hunk 8]
```diff
@@ -0,0 +1,106 @@
+#include "mbed.h"
+#include "mbed_trace.h"
+
+const char* cert = \
+    "-----BEGIN CERTIFICATE-----\n"\
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: None of these end-of-line `\` are necessary.

File: TLSSocket/main.cpp [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+        result = socket->send(query+result, size);
+        if (result < 0) {
+            printf("Error! socket->send() returned: %d\n", result);
+            goto DISCONNECT;
+        }
+        size -= result;
+    }
+
+    // Receieve an HTTP response and print out the response line
+    while (0 < (result = socket->recv(buffer, 255))) {
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: Backwards comparisons irritate me.

File: TLSSocket/main.cpp [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+    printf("Connecting to network\n");
+    result = net->connect();
+    if (result != 0) {
+        printf("Error! net->connect() returned: %d\n", result);
+        return result;
+    }
+
+    TLSSocket *socket = new TLSSocket;
+    socket->set_root_ca_cert(cert);
```
Comment (kjbracey-arm) [inline TLSSocket/main.cpp]: As noted in https://github.com/ARMmbed/mbed-os/issues/8638#issuecomment-437836993, shouldn't continue after error here.

**Pull request merged**